test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -1926,17 +1931,27 @@ public synchronized String startNode(Settings settings) {
         }
         final List<NodeAndClient> nodes = new ArrayList<>();
         final int prevMasterCount = getMasterNodesCount();
+        boolean bootstrapNodeRequired = prevMasterCount == 0;
         for (Settings nodeSettings : settings) {
-            nodes.add(buildNode(nodeSettings, defaultMinMasterNodes, () -> rebuildUnicastHostFiles(nodes)));
+            final Settings nodeSettingsIncludingBootstrap;
+            if (bootstrapNodeRequired) {
+                nodeSettingsIncludingBootstrap = Settings.builder()
+                    .put(INITIAL_MASTER_NODE_COUNT_SETTING.getKey(),
+                        (int) Stream.of(settings).filter(Node.NODE_MASTER_SETTING::get).count())
+                    .put(nodeSettings)
+                    .build();
+                bootstrapNodeRequired = false;
```

Comment:
same here, why only bootstrap one node?

Reply:
Done in 150f47b6a50b434fd4d9569683508cdc1a3f5d7d.